NeuroID/NeuroIDClass/Extensions/NIDUser.swift

```diff
@@ -145,4 +145,26 @@ public extension NeuroID {
 
         return res
     }
+
+    /**
+        This should be called the moment a user trys to login. Returns true or false based on if a passed in UserID is valid.
+        @param {String} [attemptedRegisteredUserId] - an optional identifier for the login
+     */
+    static func attemptedLogin(_ attemptedRegisteredUserId:String? = nil) -> Bool{
+        var isValidID = false
+        if (attemptedRegisteredUserId != nil) {
+            if (!NeuroID.validateUserID(attemptedRegisteredUserId!)) {
+                isValidID = false
+            } else {
+                isValidID = true
+            }
+        }
+            
+        let hashedUserId = isValidID
+        ? attemptedRegisteredUserId?.hashValue() : "scrubbed-id-failed-validation"
+        
```

Comment:
Consider instead
``` 
let hashedUserId:String
if (!NeuroID.validateUserID(attemptedRegisteredUserId ?? "")) {
    hashedUserId = attemptedRegisteredUserId!.hashValue()
} else {
    hashedUserId = "scrubbed-id-failed-validation"
}
```
 so we don't have to do a null check and validation check but rather let validation do all of it. Also removes the need for the `isValidID` variable

Reply:
where would it be duplicating?

Reply:
Saving to the datastore and the return of the validation if you put it in that block. If you had another idea mind showing me ?